src/DnsClient/DnsQueryOptions.cs

```diff
@@ -219,8 +220,33 @@ public int ExtendedDnsBufferSize
         /// Gets or sets a flag indicating whether EDNS should be enabled and the <c>DO</c> flag should be set.
         /// Defaults to <c>False</c>.
         /// </summary>
-        public bool RequestDnsSecRecords { get; set; } = false;
-
+        public bool RequestDnsSecRecords { get; set; } = false;
+
+        /// <summary>
+        /// Gets or sets a flag indicating whether the DNS failures are being cached. The purpose of caching 
+        /// failures is to reduce repeated lookup attempts within a short space of time.
+        /// Defaults to <c>False</c>.
+        /// </summary>
+        public bool UseCacheForFailures { get; set; } = false;

```

Comment:
Could you rename that to `CacheFailedResults`

and the TimeSpan to `FailedResultsCacheDuration`. CacheFailureDuration sounds like we set a value for how long the cache should fail ;)